examples/gpt-2/main.cpp

```diff
@@ -703,11 +760,10 @@ bool gpt2_eval(
     ggml_allocr_alloc_graph(allocr, gf);
 
     // run the computation
-    struct ggml_cplan plan = ggml_graph_plan(gf, n_threads);
-    static std::vector<uint8_t> work_buffer;
-    work_buffer.resize(plan.work_size);
-    plan.work_data = work_buffer.data();
-    ggml_graph_compute(gf, &plan);
+    if (strcmp(ggml_backend_name(model.backend), "CPU") == 0) {
```

Comment:
Ah guess this is a way for figuring out whether a backed is CPU (my prev comment). Can we add a helper function returning the type of backend ?